torch/csrc/jit/serialization/export.cpp

```diff
@@ -864,6 +878,30 @@ unsigned long long int GraphEncoder::GetGraphProtoSize(
     size += tensor_proto->ByteSizeLong();
     size += tensor.element_size() * tensor.numel();
   }
+
+  // Add up all onnx::Constant nodes that are Tensors
+  for (const auto& node : graph->nodes()) {
+    if (node->kind() == ::c10::onnx::Constant &&
+        node->hasAttribute(attr::value) &&
+        node->kindOf(attr::value) == AttributeKind::t) {
+      at::Tensor tensor = node->t(attr::value);
+
+      // Don't actually copy the buffer into n_proto since that is expensive.
+      // All we actually need is its size.
+      auto* n_proto = graph_proto_copy.add_node();
+      EncodeNode(
```

Comment:
@thiagocrepaldi are you sure this is not double counting the tensor size since n_proto actually has the tensor copied as attribute?

Besides this is affecting the resulting node name, maybe just keep the tensor addition below is better.

Reply:
The repro had a single initializer (500MB) and 5.3GB in multiple constants, but `graph_proto_copy.ByteSizeLong()` still didn't detect the > 2GB